File: Leetcode/Java/ReverseStringArray.java
/* Given an array of chars {'h', 'e', 'l', 'l', 'o'}
 * reverse the array in place without creating
 * another array in memory
 * example output: {'o', 'l', 'l', 'e', 'h'}
 */

class Solution {
    public static void main(Char[] Array) {
        char temp;
        for (int i = 0; i < Array.length / 2; i++) {
            temp = Array[i];
            Array[i] = Array[Array.length - i - 1];
            Array[Array.length - i - 1] = temp;
        }
        System.out.println(Array);
    }
}

class BlazinglyFastSolution {
    public void reverseString(char[] s) {
        int i = 0;
        int j = s.length - 1;
        while (i < j) {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            i++;
            j--;
        }
    }
}

File: Leetcode/Python/ReverseStringArray.py



class DumbSolution:
    def reverseString(self, s:List[str]) -> None:
        s.reverse()

class SimpleSolution:
    def reverseString(self, s: List[str]) -> None:
        s[:] = s[::-1]
        

class SplitSolution:
    def reverseString(self, s: List[str]) -> None:
        l,r=0,len(s)-1
        while l<r:
            s[l],s[r]=s[r],s[l]
            l,r=l+1,r-1

class Solution:
    def reverseString(self, s: List[str]) -> None:

        tmp=''
        i=0
        while (i < len(s)/2):
            tmp = s[i]
            s[i] = s[len(s)-i-1]
            s[len(s)-i-1] = tmp
            i+=1
